docs/apps/hyperqueue.md

```diff
@@ -0,0 +1,241 @@
+    resources.
+
+HyperQueue (HQ) is a tool for efficient sub-node task scheduling. Instead of submitting each one of
+your computational tasks using `sbatch` or `srun` you can instead allocate a large resource block
+and then use HyperQueue to submit your tasks. This will be much less stressful for the batch queue
+system and is therefore the recommended way to run your high-throughput computing use cases.
+
+## License
+
+Free to use and open source under [MIT License](https://github.com/It4innovations/hyperqueue/blob/main/LICENSE)
+
+## Available
+
+* Puhti: 0.4.0, 0.9.0
+* Mahti: 0.11-dev
+
+## Usage
+
+``
+module load hyperqueue
+``
+
+HyperQueue works on a worker-server-client basis. Here, the server manages connections, workers are
+started on compute nodes and execute commands which the client submitted to the server. This
+resembles a Slurm within a Slurm, but you have to start the server and workers yourself.
+
+### Starting the server
+
+!!! info "Note"
+    The following instructions apply for Slurm job scripts where only full nodes
+    are allocated. [A full example](#full-example) (for Mahti) can be found at the bottom.
+
+Specify where the HyperQueue server should be placed. All `hq` commands respect this variable
+so make sure it's set before you call any `hq` commands. The server location can also be placed
+using the command line flag `--server-dir /server/location`.
+
+```bash
+export HQ_SERVER_DIR=/server/location
+```
+
+If the server directory is not specified it will default to the user home directory. In this case
+one has to be careful not to mix up separate computations. For simple cases that fit inside one
+Slurm job, we recommend starting one server per job in some job-specific directory.
+
+Start the server:
+
+```bash
+hq server start & 
+until hq jobs ; do sleep 1 ; done
+```
+
+Here, the server is placed in the background (`&`) so that we can continue.
+
+### Starting the workers
+
+Start the workers (again in the background so our script can continue):
+
+```bash
+srun --cpu-bind=none --hint=nomultithread --mpi=none --ntasks-per-node=$SLURM_NNODES -c $SLURM_CPUS_PER_TASK hq worker start --cpus=$SLURM_CPUS_PER_TASK &
+```
+
+Here, we launch one worker per node with each worker getting the full node. If you need HQ
+to be aware of other resources, e.g memory, local disk or GPUS, see the [Generic resource
+section](https://it4innovations.github.io/hyperqueue/v0.11.0/jobs/gresources/) in the
+official documentation.
+
+Next, we can start submitting jobs or alternatively wait for all the workers to connect
+to the server before submission. This is generally good practice as we can notice issues with
+the workers early.
+
+Loop until all workers are online (note no timeout):
+
+```bash
+while true; do
+
+    num_up=$(hq worker list | grep -c RUNNING)
+    echo "Checking if workers have started"
+    if [[ $num_up -eq $SLURM_NNODES ]]; then
```

Comment:
Honestly this `if` should be the loop condition instead of promoting `while true` but it might be hard to read...? 🤔